src/vscode-dts/vscode.proposed.terminalCompletionProvider.d.ts

```diff
@@ -16,10 +16,9 @@ declare module 'vscode' {
 		 * @param token A cancellation token.
 		 * @return A list of completions.
 		 */
-		provideTerminalCompletions(terminal: Terminal, context: TerminalCompletionContext, token: CancellationToken): ProviderResult<T[]>;
+		provideTerminalCompletions(terminal: Terminal, context: TerminalCompletionContext, token: CancellationToken): ProviderResult<T[] | TerminalCompletionList<T>>;
```

Comment:
discussed this in the API sync - models https://github.com/microsoft/vscode/blob/f6d4a459f93868b9bf8d920c6be3860c775c090e/src/vscode-dts/vscode.d.ts#L5131